### DISCUSSION TITLE
无法显示EMV数据图形

### DISCUSSION THREAD

**Discussion by tangyixiang** [klinecharts/KLineChart#422] [Q&A]
用下面的数据的数据，无法绘制EMV图形，找不到原因
`[{"open":4.9,"low":4.82,"close":4.99,"high":5.06,"timestamp":1609718400000,"volume":756579,"turnover":375471664},{"open":4.95,"low":4.91,"close":5.13,"high":5.18,"timestamp":1609804800000,"volume":831861,"turnover":422081008},{"open":5.22,"low":4.78,"close":4.87,"high":5.24,"timestamp":1609891200000,"volume":769114,"turnover":378967472},{"open":4.8,"low":4.38,"close":4.38,"high":4.81,"timestamp":1609977600000,"volume":1068727,"turnover":477241328},{"open":4.3,"low":4.11,"close":4.29,"high":4.35,"timestamp":1610064000000,"volume":548959,"turnover":232893036},{"open":4.37,"low":4.19,"close":4.25,"high":4.48,"timestamp":1610323200000,"volume":461419,"turnover":199014760},{"open":4.2,"low":4.12,"close":4.2,"high":4.28,"timestamp":1610409600000,"volume":356543,"turnover":149024420},{"open":4.16,"low":3.98,"close":4.02,"high":4.16,"timestamp":1610496000000,"volume":406078,"turnover":163995259},{"open":4.03,"low":3.96,"close":4.07,"high":4.1,"timestamp":1610582400000,"volume":222339,"turnover":89487979},{"open":4.05,"low":4.03,"close":4.11,"high":4.14,"timestamp":1610668800000,"volume":164018,"turnover":67217253},{"open":4.1,"low":4.09,"close":4.09,"high":4.17,"timestamp":1610928000000,"volume":205209,"turnover":84675108},{"open":4.06,"low":4,"close":4.01,"high":4.08,"timestamp":1611014400000,"volume":220275,"turnover":89011003},{"open":4.02,"low":3.99,"close":4.11,"high":4.12,"timestamp":1611100800000,"volume":189297,"turnover":76759564},{"open":4.1,"low":4.07,"close":4.24,"high":4.34,"timestamp":1611187200000,"volume":403233,"turnover":170797387},{"open":4.21,"low":4.11,"close":4.14,"high":4.22,"timestamp":1611273600000,"volume":231642,"turnover":96103141},{"open":4.13,"low":4.03,"close":4.08,"high":4.16,"timestamp":1611532800000,"volume":206961,"turnover":84850786},{"open":4.04,"low":3.99,"close":4.01,"high":4.09,"timestamp":1611619200000,"volume":177085,"turnover":71304395},{"open":3.99,"low":3.95,"close":4.01,"high":4.03,"timestamp":1611705600000,"volume":140828,"turnover":56228346},{"open":3.99,"low":3.95,"close":3.98,"high":4.03,"timestamp":1611792000000,"volume":142455,"turnover":56772925},{"open":3.98,"low":3.72,"close":3.76,"high":4,"timestamp":1611878400000,"volume":283057,"turnover":108821146},{"open":3.75,"low":3.65,"close":3.72,"high":3.83,"timestamp":1612137600000,"volume":193400,"turnover":72421337},{"open":3.75,"low":3.7,"close":3.78,"high":3.78,"timestamp":1612224000000,"volume":137692,"turnover":51622527},{"open":3.78,"low":3.67,"close":3.68,"high":3.78,"timestamp":1612310400000,"volume":150096,"turnover":55833783},{"open":3.67,"low":3.5,"close":3.55,"high":3.68,"timestamp":1612396800000,"volume":219533,"turnover":78599446},{"open":3.56,"low":3.48,"close":3.5,"high":3.63,"timestamp":1612483200000,"volume":114541,"turnover":40549083},{"open":3.48,"low":3.35,"close":3.47,"high":3.53,"timestamp":1612742400000,"volume":134882,"turnover":46767645},{"open":3.46,"low":3.41,"close":3.51,"high":3.53,"timestamp":1612828800000,"volume":114976,"turnover":40041305},{"open":3.51,"low":3.51,"close":3.63,"high":3.67,"timestamp":1612915200000,"volume":156976,"turnover":56580624},{"open":3.74,"low":3.69,"close":3.78,"high":3.8,"timestamp":1613606400000,"volume":177737,"turnover":66761807},{"open":3.8,"low":3.76,"close":3.92,"high":3.92,"timestamp":1613692800000,"volume":182106,"turnover":70332875}]`

**Comment by liihuu** [comment]
应该是计算出来的数值太小了，精度不够，不过计算公式也应该优化一下。

**Reply by tangyixiang** [reply]
好的，感谢

**Comment by liihuu** [comment]
#425
